jre_emul/Classes/com/google/j2objc/security/IosRSAKeyFactory.java

```diff
@@ -52,17 +52,17 @@ protected PublicKey engineGeneratePublic(KeySpec keySpec) throws InvalidKeySpecE
   protected PrivateKey engineGeneratePrivate(KeySpec keySpec)
       throws InvalidKeySpecException {
 	  //The KeySpec for Private Key is PKCS8
-	 if (keySpec instanceof PKCS8EncodedKeySpec ) {
-	     return new IosRSAKey.IosRSAPrivateKey(((PKCS8EncodedKeySpec) keySpec).getEncoded());  
-	 }
-    if (keySpec instanceof X509EncodedKeySpec || keySpec instanceof PKCS8EncodedKeySpec ) {
-      X509EncodedKeySpec x509Spec = (X509EncodedKeySpec) keySpec;
-      return new IosRSAKey.IosRSAPrivateKey(x509Spec.getEncoded());
+	if (keySpec instanceof PKCS8EncodedKeySpec ) {
+	    return new IosRSAKey.IosRSAPrivateKey(((PKCS8EncodedKeySpec) keySpec).getEncoded());  
+	}
+    if (keySpec instanceof X509EncodedKeySpec) {
+        X509EncodedKeySpec x509Spec = (X509EncodedKeySpec) keySpec;
+        return new IosRSAKey.IosRSAPrivateKey(x509Spec.getEncoded());
     } else if (keySpec instanceof RSAPrivateKeySpec) {
-      return new IosRSAKey.IosRSAPrivateKey((RSAPrivateKeySpec) keySpec);
+        return new IosRSAKey.IosRSAPrivateKey((RSAPrivateKeySpec) keySpec);
     }
     throw new InvalidKeySpecException(
-        "Must use PKCS8EncodedKeySpec; was " + keySpec.getClass().getName());
+        "Must use PKCS8EncodedKeySpec, X509EncodedKeySpec or RSAPrivateKeySpec; was " + keySpec.getClass().getName());
```

Comment:
Please split line as follows, to avoid exceeding 100 column limit:
```
       "Must use PKCS8EncodedKeySpec, X509EncodedKeySpec or RSAPrivateKeySpec; was "
           + keySpec.getClass().getName());
```